addons/point_of_sale/models/point_of_sale_category.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from openerp import api, fields, models, tools
+
+
+class PosCategory(models.Model):
+    _name = "pos.category"
+    _description = "Public Category"
+    _order = "sequence, name"
+
+    _constraints = [
+        (models.Model._check_recursion, 'Error ! You cannot create recursive categories.', ['parent_id'])
+    ]
+
+    @api.multi
+    def name_get(self):
+        res = []
+        for cat in self:
+            names = [cat.name]
+            pcat = cat.parent_id
+            while pcat:
+                names.append(pcat.name)
+                pcat = pcat.parent_id
+            res.append((cat.id, ' / '.join(reversed(names))))
+        return res
+
+    @api.multi
+    def _name_get_fnc(self):
+        for cat in self:
+            cat.complete_name = dict(cat.name_get()).values()[0]
+
+    @api.depends('image')
+    def _get_image(self):
+        for cat in self:
+            cat.image_medium = tools.image_get_resized_images(cat.image)
+
+    @api.one
+    def _set_image(self):
+        return self.write({'image': tools.image_resize_image_big(self.image)})
+
+    name = fields.Char(required=True, translate=True)
+    complete_name = fields.Char(compute='_name_get_fnc', string='Name')
```

Comment:
I think, we should override display_name field. also deprecate complete_name field